src/pcrlock/pcrlock.c

```diff
@@ -2691,6 +2691,103 @@ static int make_pcrlock_record(
         return 0;
 }
 
+static void evp_md_ctx_free_all(EVP_MD_CTX **md[]) {
+        assert(md);
+
+        if (!*md)
+                return;
+
+        for (size_t i = 0; (*md)[i]; i++)
+                EVP_MD_CTX_free((*md)[i]);
+
+        *md = mfree(*md);
+}
+
+static int make_pcrlock_record_from_stream(
+                uint32_t pcr,
+                FILE *f,
+                JsonVariant **ret_record) {
+        _cleanup_(json_variant_unrefp) JsonVariant *digests = NULL;
+        _cleanup_(evp_md_ctx_free_all) EVP_MD_CTX **mdctx = NULL;
```

Comment:
Could be stored on the stack + `static void evp_md_ctx_free_all(EVP_MD_CTX *(*md)[TPM2_N_HASH_ALGORITHMS])`

Reply:
In that case the size would not be known, and inside the `evp_md_ctx_free_all` I would need to iter until `TPM2_N_HASH_ALGORITHMS`.  I was not able to use `sizeof()` because array size is lost in C function arguments.  Somehow I found this ugly.

Reply:
> In that case the size would not be known

If you use `static void evp_md_ctx_free_all(EVP_MD_CTX *(*md)[TPM2_N_HASH_ALGORITHMS])`, it is known and you can do the usual `sizeof(*md)/sizeof(**md)` dance. The compiler will also warn resp. error out if the array in the function parameter has the wrong size.

Reply:
Very cool. Done!